src/torrent.ts

```diff
@@ -201,15 +206,40 @@ export async function getTorrentByFuzzyName(
 	name: string
 ): Promise<null | Metafile> {
 	const allNames = await db("torrent").select("name", "file_path");
-	// @ts-expect-error fuse types are confused
-	const potentialMatches = new Fuse(allNames, {
-		keys: ["name"],
-		distance: 6,
-		threshold: 0.25,
-	}).search(name);
-
-	if (potentialMatches.length === 0) return null;
-	const [firstMatch] = potentialMatches;
+
+	const fullMatch = reformatTitleForSearching(name).replace(
+		/[^a-z0-9]/gi,
+		""
+	);
+
+	// Attempt to filter torrents in DB to match incoming torrent before fuzzy check
+	let filteredNames = [];
+
+	if (fullMatch) {
+		filteredNames = allNames.filter((dbName) => {
+			const dbMatch = reformatTitleForSearching(dbName.item.name).replace(
```

Comment:
where is this `.item` coming from? `allNames` looks like it has a shape of `{ name: string, file_path: string }[]`

Reply:
copied from below, `return parseTorrentFromFilename(firstMatch.item.file_path);`

not sure where it comes from. It's part of the syntax I'm not familiar with, but can change it

Reply:
I think that's part of the Fuse api